backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Flatrate_Term.java

```diff
@@ -87,56 +82,34 @@ public void createInterimContractIfNeeded(@NonNull final I_C_Flatrate_Term flatr
 	}
 
 	@DocValidate(timings = ModelValidator.TIMING_AFTER_COMPLETE)
-	public void createInterimContractLogs(@NonNull final I_C_Flatrate_Term flatrateTermRecord)
+	public void onInterimContractComplete(@NonNull final I_C_Flatrate_Term flatrateTermRecord)
 	{
 		if (!TypeConditions.ofCode(flatrateTermRecord.getType_Conditions()).isInterimContractType())
 		{
 			return;
 		}
+		trxManager.runAfterCommit(() -> createInterimContractLogs(flatrateTermRecord));
```

Comment:
@mrie04  why after commit?

Reply:
Because we have a sysConfig for automatically creating Interim Contracts on modular Contract Complete (which all is done on order complete) and without it the InterimContractHandler couldn't find the needed Modular Contract data.